sdk/tables/azure-data-tables/src/main/java/com/azure/data/tables/implementation/TablesJacksonSerializer.java

```diff
@@ -75,7 +76,9 @@ public void serialize(Object object, SerializerEncoding encoding, OutputStream o
 
     @Override
     public <U> U deserialize(String value, Type type, SerializerEncoding serializerEncoding) throws IOException {
-        if (type == TableEntityQueryResponse.class) {
+        if (type == TableEntityQueryResponse.class
+            || (type instanceof ParameterizedType) && ((ParameterizedType) type).getRawType() == Map.class) {
```

Comment:
I don't think this is needed. If the type is Map, then Jackson should handle deserializing it just fine without the custom logic.

Reply:
If we don't do this then Jackson deserializes some Map's values to String instead of the odata type specified. I found out about this with tests that failed when deserializing byte arrays and OffsetDateTime, for example. Maybe there's a better way to check for this, so I'm holding off on merging these changes until we have a chat on Monday :)